cli/src/commands/status.rs

```diff
@@ -98,6 +99,9 @@ pub(crate) fn cmd_status(
         let ancestors_conflicts = RevsetExpression::filter(RevsetFilterPredicate::HasConflict)
             .intersection(&wc_revset.ancestors())
             .minus(&wc_revset)
+            .minus(&revset_util::parse_immutable_expression(
+                &workspace_command.revset_parse_context(),
+            )?)
```

Comment:
nit: Since `immutable_heads()` is a user expression, `.evaluate_programmatic()` can't be used anymore.